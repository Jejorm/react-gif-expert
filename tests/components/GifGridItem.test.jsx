import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe( 'Pruebas en <GifGridItem />', () => {

    const title = 'New Title'
    const url = 'https://gifimage.jpg'

    test( 'Debe hacer match con el snapshot', () => {

        const { container } = render( <GifGridItem title={ title } url={ url } /> )

        expect( container ).toMatchSnapshot()
    } )

    test( 'Debe mostrar la imagen con el URL y ALT indicado', () => {

        render( <GifGridItem title={ title } url={ url } /> )

        const { src, alt } = screen.getByRole( 'img' )

        expect( src ).toContain( url )
        expect( alt ).toBe( title )
    } )

    test( 'Debe mostrar el tÃ­tulo en el componente', () => {

        render( <GifGridItem title={ title } url={ url } /> )

        expect( screen.getByText( title ) ).toBeTruthy()
    } )
} )