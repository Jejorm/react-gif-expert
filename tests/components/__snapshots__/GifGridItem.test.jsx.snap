// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="New Title"
      src="https://gifimage.jpg"
    />
    <p>
      New Title
    </p>
  </div>
</div>
`;
